### PR TITLE
net: openthread: rpc: SRP client's RPC client

### PR DESCRIPTION
**Pull request by konradderda**

### CODE REVIEW

**Review by alxelax — CHANGES_REQUESTED** [review]

File: tests/subsys/net/openthread/rpc/client/src/srp_client_suite.c [hunk 1]
```diff
@@ -387,4 +412,129 @@ ZTEST(ot_rpc_srp_client, test_otSrpClientSetTtl)
...
+{
+	mock_nrf_rpc_tr_expect_add(RPC_CMD(OT_RPC_CMD_SRP_CLIENT_STOP), RPC_RSP());
+	otSrpClientStop(NULL);
+	mock_nrf_rpc_tr_expect_done();
+}
+
+/* Test serialization of otSrpClientIsRunning() */
+ZTEST(ot_rpc_srp_client, test_otSrpClientIsRunning)
+{
+	otError err_1st;
```
Comment (Damian-Nordic) [inline tests/subsys/net/openthread/rpc/client/src/srp_client_suite.c:443]: Why is `otError` used to store bool?

File: tests/subsys/net/openthread/rpc/client/src/srp_client_suite.c [hunk 2]
```diff
@@ -387,4 +412,129 @@ ZTEST(ot_rpc_srp_client, test_otSrpClientSetTtl)
...
+	error = otSrpClientSetHostAddresses(NULL, addrs, 4);
+	mock_nrf_rpc_tr_expect_done();
+
+	zassert_equal(error, OT_ERROR_NONE);
+}
+
+/* Test serialization of otSrpClientIsAutoStartModeEnabled() */
+ZTEST(ot_rpc_srp_client, test_otSrpClientIsAutoStartModeEnabled)
+{
+	otError err1;
```
Comment (Damian-Nordic) [inline tests/subsys/net/openthread/rpc/client/src/srp_client_suite.c:497]: ditto

File: tests/subsys/net/openthread/rpc/common/test_rpc_env.h [hunk 3]
```diff
@@ -34,6 +34,8 @@
 #define RPC_RSP(...)      RPC_PKT(0x01, 0xff, 0x00, 0x00, 0x00 __VA_OPT__(,) __VA_ARGS__, 0xf6)
 #define NO_RSP            RPC_PKT()
 
+#define CBOR_LIST(...) 0x9F, __VA_ARGS__, 0xFF
```
Comment (Damian-Nordic) [inline tests/subsys/net/openthread/rpc/common/test_rpc_env.h:37]: Could you move this to "/* Macros for constructing CBOR data items. */" section above?

**Review by Damian-Nordic — COMMENTED** [review]
Some extra comments from me

File: samples/nrf_rpc/protocols_serialization/client/src/ot_shell.c [hunk 4]
```diff
@@ -15,12 +15,63 @@
...
+			return &services[i];
+		}
+	}
+
+	return NULL;
+}
+
+static void service_free(otSrpClientService *service)
+{
+	for (int i = 0; i < SERVICE_NUM; ++i) {
```
[the remark this answers is not in the document]
Comment (alxelax) [reply samples/nrf_rpc/protocols_serialization/client/src/ot_shell.c]: I meant that if you use `IS_ARRAY_ELEMENT` then you do not need a cycle at all. Code will be much optimal both speed and size.

```
	if (IS_ARRAY_ELEMENT(services, service)) {
                struct srp_service *elem = CONTAINER_OF(service, struct srp_service, service);
                
                memset(elem, 0, sizeof(struct srp_service));   
	}
```

**Review by alxelax — COMMENTED** [review]
Some comments have not been resolved.

**Review by canisLupus1313 — APPROVED** [review]

File: samples/nrf_rpc/protocols_serialization/client/src/ot_shell.c [hunk 5]
```diff
@@ -738,6 +789,470 @@ static int cmd_ot(const struct shell *sh, size_t argc, char *argv[])
...
+	return 0;
+}
+
+static int cmd_test_srp_client_host_name(const struct shell *sh, size_t argc, char *argv[])
+{
+	strncpy(srp_client_host_name, argv[1], sizeof(srp_client_host_name));
+
+	otSrpClientSetHostName(NULL, srp_client_host_name);
+
+	shell_print(sh, "Name set to: %s", argv[1]);
```
[the remark this answers is not in the document]
Comment (Damian-Nordic) [reply samples/nrf_rpc/protocols_serialization/client/src/ot_shell.c:850]: But argv must be null-terminated, no?

**Review by Damian-Nordic — APPROVED** [review]

**Review by alxelax — APPROVED** [review]

**Pull request merged**